File: ShallowWaterSolvers/incompressible/rossbywave/cacheCeta.cpp
#include "nr.h"
#include "parameters.h"

void NR::cacheCeta(Mat_O_DP &Ceta)
{
	// Computes the values of the cosine eta basis
	// functions at each collocation point.

	DP deleta = pi/(M*kappa);
	DP eps = deleta/2.0;
	for (int m=1;m<=M;m++)
	{
		for (int i=1;i<=M;i++)
		{
			Ceta[m-1][i-1]=cos((deleta*(i-1)+eps)*m*kappa);
		}
	}
}

File: Results/Plots/Fourier Series methods/old code/Different truncation methods/Truncation 3/cacheCeta.m
function Ceta = cacheCeta(M,kappa)

%// Computes the values of the cosine basis
%// functions at each collocation point.

Ceta=zeros(M,M);

deleta = pi/(M*kappa);
eps=deleta/2;
for m=1:M
	for i=1:M
		Ceta(m,i)=cos((deleta*(i-1)+eps)*m*kappa);
    end
end
